From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Saifuddin Hitawala <hitawala@chromium.org>
Date: Thu, 20 Feb 2025 09:23:32 -0800
Subject: [gpu] Remove unnecessary media_buildflags include

gpu_preferences.h includes media_buildflags.h which is not really
needed. This change removes the include and includes it in
GpuServiceImpl where it is really needed.

This helps fix a build failure due to transitive dependency on
media_buildflags.h from gpu_info.h via gpu_preferences.h

Bug: 397776075
Change-Id: I86abb6f779b416e89fa79b84c319e40cf8bc18c9
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/6286526
Commit-Queue: Saifuddin Hitawala <hitawala@chromium.org>
Reviewed-by: Kyle Charbonneau <kylechar@chromium.org>
Cr-Commit-Position: refs/heads/main@{#1422628}

diff --git a/components/viz/service/BUILD.gn b/components/viz/service/BUILD.gn
index f53aa0c2c8b3a428c08ca6715d6b837a28ad0892..9cb52cb768e648235f89f65e870d853d45268900 100644
--- a/components/viz/service/BUILD.gn
+++ b/components/viz/service/BUILD.gn
@@ -295,6 +295,7 @@ viz_component("service") {
     "//gpu/skia_bindings:skia_bindings",
     "//gpu/vulkan:buildflags",
     "//media",
+    "//media:media_buildflags",
     "//media/capture:capture_lib",
     "//media/mojo/services",
     "//services/metrics/public/mojom",
diff --git a/components/viz/service/gl/DEPS b/components/viz/service/gl/DEPS
index 2da7495c74342705bd9511aef880cda09ea09f6e..27fc6ebab8e83eeaebcbc5ebf9ce151120039362 100644
--- a/components/viz/service/gl/DEPS
+++ b/components/viz/service/gl/DEPS
@@ -18,6 +18,7 @@ include_rules = [
   "+media/base/media_log.h",
   "+media/base/media_switches.h",
   "+media/base/win/mf_feature_checks.h",
+  "+media/media_buildflags.h",
   "+media/gpu",
   "+media/mojo",
   "+mojo/public/cpp",
diff --git a/components/viz/service/gl/gpu_service_impl.h b/components/viz/service/gl/gpu_service_impl.h
index 2da100053e70bf3dcea83d17359938258a10c5bd..b399263bf6780fdb6a2082487f3a6e0d3fed8704 100644
--- a/components/viz/service/gl/gpu_service_impl.h
+++ b/components/viz/service/gl/gpu_service_impl.h
@@ -42,6 +42,7 @@
 #include "gpu/ipc/service/gpu_config.h"
 #include "gpu/ipc/service/x_util.h"
 #include "gpu/vulkan/buildflags.h"
+#include "media/media_buildflags.h"
 #include "mojo/public/cpp/bindings/pending_receiver.h"
 #include "mojo/public/cpp/bindings/receiver.h"
 #include "mojo/public/cpp/bindings/receiver_set.h"
diff --git a/gpu/config/BUILD.gn b/gpu/config/BUILD.gn
index a310110d051b8daccd76b72e6b91c9e39dca19db..dc6eca920300a4ed486be3c124e064196b16e3d5 100644
--- a/gpu/config/BUILD.gn
+++ b/gpu/config/BUILD.gn
@@ -194,7 +194,6 @@ source_set("config_sources") {
     "//gpu/command_buffer/common:common_sources",
     "//gpu/ipc/common:gpu_preferences_interface",
     "//gpu/vulkan:buildflags",
-    "//media:media_buildflags",
     "//services/webnn/public/mojom",
     "//skia:buildflags",
     "//third_party/re2",
diff --git a/gpu/config/DEPS b/gpu/config/DEPS
index b0b037e2572196086e86874eb01677a467eb81b7..82511c40c7f5f2af7bcacc0fdc11f91265ed736f 100644
--- a/gpu/config/DEPS
+++ b/gpu/config/DEPS
@@ -1,7 +1,6 @@
 include_rules = [
   "+components/crash/core/common/crash_key.h",
   "+components/miracle_parameter",
-  "+media/media_buildflags.h",
   "+services/webnn/public/mojom/features.mojom-features.h",
   "+skia/buildflags.h",
   "+third_party/dawn/include",
diff --git a/gpu/config/gpu_preferences.h b/gpu/config/gpu_preferences.h
index 4cfc206e30fbc79d9c57bb81c46ee58103a47686..b9241b4bff3556fb312b017e4c50ea93a42d3554 100644
--- a/gpu/config/gpu_preferences.h
+++ b/gpu/config/gpu_preferences.h
@@ -11,7 +11,6 @@
 
 #include "build/build_config.h"
 #include "gpu/gpu_export.h"
-#include "media/media_buildflags.h"
 #include "ui/gfx/buffer_types.h"
 
 #if BUILDFLAG(IS_OZONE)
